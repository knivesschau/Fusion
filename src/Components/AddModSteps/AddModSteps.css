/* mobile styles first, min 320px */
#add-mod-step {
    font-size: .90rem;
    margin-top: 1.5em;
}

#add-mod-step {
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#remove-mod-step {
    background-color: #fac59d;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#add-mod-step:hover, #remove-mod-step:hover {
    cursor: pointer; 
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#add-mod-step:active, #remove-mod-step:active {
    background-color: #881600;
}

#remove-mod-step {
    margin-top: .25em;
    margin-bottom: .25em;
}

#mod-steps-add {
    vertical-align: middle;
    width: 80%;
    margin: .25em;
}

/* tablet and desktop styles */ 
@media only screen and (min-width: 768px) {
    #add-mod-step {
        font-size: 1.25rem;
        padding-top: .35em;
        padding-bottom: .35em;
        margin-top: .25em;
    }
    
    #remove-mod-step {
        font-size: 1rem;
        margin: 1em;
        padding-top: .25em;
        padding-bottom: .25em;
    }

    #mod-steps-add {
        font-size: 1.10rem;
    }
}

@media only screen and (min-width: 1200px) {
    #add-mod-step {
        font-size: 1.45rem;
    }
    
    #remove-mod-step {
        font-size: 1.15rem;
    }

    #mod-steps-add {
        font-size: 1.25rem;
    }
} 