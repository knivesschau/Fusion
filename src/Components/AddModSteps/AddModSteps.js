import React, { useState } from 'react';
import './AddModSteps.css';

export default function AddModSteps() {
    const [modSteps, setModSteps] = useState({ step: [] });

    function createInputs() {
        return modSteps.step.map((step, i) => 
            <li id="add-mod-steps" key={i}>
                <textarea
                    rows="10"
                    id="mod_steps"
                    name="mod_steps"
                    value={step || ""}
                    onChange={handleUpdate.bind(i)}/>

                <button type="button" id="remove-mod-ingredient" onClick={removeInput.bind({i})}>Remove</button>
            </li>
        );
    };
    
    function handleUpdate(e) {
        let newSteps = [...modSteps.step];
        newSteps[this] = e.target.value;

        setModSteps({
            step: newSteps
        });
    };

    const addInput = () => {
        setModSteps({
            step: [...modSteps.step, ""]
        });
    };

    const removeInput = () => {
        let newSteps = [...modSteps.step];
        newSteps.splice(this, 1);

        setModSteps({
            step: newSteps
        });
    };

    return (
        <>
        {createInputs()}
        <button type="button" id="add-mod-stept" onClick={addInput}>Add Ingredient</button>
        </>
    );
};
