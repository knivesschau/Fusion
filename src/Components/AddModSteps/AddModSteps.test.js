import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AddModSteps from './AddModSteps';

describe ('AddModSteps Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddModSteps/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete AddModSteps Component', () => {
        const wrapper = shallow(<AddModSteps/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});