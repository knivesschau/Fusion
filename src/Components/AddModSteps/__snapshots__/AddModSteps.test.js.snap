// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddModSteps Component Renders the complete AddModSteps Component 1`] = `
<Fragment>
  <button
    id="add-mod-stept"
    onClick={[Function]}
    type="button"
  >
    Add Step
  </button>
</Fragment>
`;
